warehouse/templates/email/token-compromised-leak/body.html

```diff
@@ -0,0 +1,68 @@
+{#
+ # Licensed under the Apache License, Version 2.0 (the "License");
+ # you may not use this file except in compliance with the License.
+ # You may obtain a copy of the License at
+ #
+ # http://www.apache.org/licenses/LICENSE-2.0
+ #
+ # Unless required by applicable law or agreed to in writing, software
+ # distributed under the License is distributed on an "AS IS" BASIS,
+ # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ # See the License for the specific language governing permissions and
+ # limitations under the License.
+-#}
+{% extends "email/_base/body.html" %}
+
+{% set site_name=request.registry.settings["site.name"] %}
+
+{% block content %}
+<h3>{% trans %}What?{% endtrans %}</h3>
+<p>{% trans site_name=site_name %}A third party ({{ origin }})
+has informed us that a {{ site_name }} API token associated
+with your account "{{ username }}" was found to be accessible at the following
+public url. We have revoked this token.{% endtrans %}</p>
+
+<p><a href="{{ public_url }}">{{ public_url }}</a></p>
+
+<h3>{% trans %}What should I do?{% endtrans %}</h3>
+<p>
+  {% trans %}First, we recommend that you inspect the public page where the token
+  appeared. Please investigate whether this is the result of an error on your side or if
+  someone else acquired and published your token.{% endtrans %}
+</p>
+<p>
+  {% trans site_name=site_name %}
+  If the token was not published by you, it means it was stolen, either from
+  you or from {{ site_name }}. We urge you to take all the necessary steps to secure all your
+  credentials.{% endtrans %}
+  {% trans site_name=site_name, href="https://python-security.readthedocs.io/packages.html#vulnerabilites-in-the-package-index" %}
+  In the case of a breach of the {{ site_name }} database, you will find informations
+  and support in the <a href="{{ href }}">python-security</a> documentation.{% endtrans %}
+  {# TODO how much of an effort should we do to replace PyPI with {{ site_name }} ? #}
```

Comment:
If we're talking about a specific site, e.g. "Your PyPI|TestPyPI token was leaked" or linking to something on the site (which should done via `request.help_url` or something similar), then we should make every effort.

I think the only exception would be the admin email, which should always be `@pypi.org`.

Reply:
Ok, I’llnkeep that in mind thanks !